### PR TITLE
Nc news 7

### PR DESCRIPTION
**Pull request by royr5**

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -109,6 +109,76 @@ describe("/api/articles", () => {
...
+  });
+  test("POST:404 sends an appropriate status and error message when given a valid but non-existent user", () => {
+    const newComment = {
+      body: "Delicious crackerbreads",
+      username: "testUser",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+
```
Comment (two-triangles) [inline __tests__/index.test.js]: The rest of this test is missing?

File: models/comments.model.js [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
+};
+
+
+
+exports.checkUserExists = (user) => {
+  return db
+    .query(`SELECT * FROM users WHERE username = $1`, [user])
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({ status: 404, msg: "path not found" });
```
Comment (two-triangles) [inline models/comments.model.js]: If the user doesn't exist when you are trying to post a comment, postgres will throw an error and you can use that status code to return an appropriate error

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+} = require("../models/comments.model");
+const { selectComments } = require("../models/comments.model");
+
+exports.postComment = (req, res, next) => {
+  const id = req.params.article_id;
+  const newComment = req.body;
+  const user = req.body.username;
+
+  Promise.all([
+    selectSingleArticle(id),
```
Comment (nickkagame) [inline controllers/comments.controller.js]: These separate check if exists are not necessary as the ID and the user are foreign keys on the comments table, and will result in PSQL errors if they don't exist.

**Review by nickkagame — APPROVED** [review]
Good work Roshan! Only minor comments

**Pull request merged**